helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -432,6 +481,17 @@ protected Map<String, ResourceAssignment> emergencyRebalance(
 
     // Step 3: persist result to metadata store
     persistBestPossibleAssignment(newAssignment);
+
+    // Step 4: handle delayed rebalance minActiveReplica
+    // Note this result is one step branching from the main calculation and SHOULD NOT be persisted -- it is temporary,
+    // and only apply during the delayed window of those offline yet active nodes, a definitive resolution will happen
+    // once the node comes back of remain offline after the delayed window.
+    Map<String, ResourceAssignment> assignmentWithDelayedRebalanceAdjust = newAssignment;
+    if (_partialRebalanceRunner.isAsyncPartialRebalanceEnabled()) {
```

Comment:
Shouldnt we be calling emergency pipeline regardless of partialRebalancRunner ?

Reply:
One correction, this is already in the emergency pipeline, it's how we should be calling partial, sync or async. The `handleDelayedRebalanceMinActiveReplica` is always called, but it has to be following the right order (after partial if sync)

Reply:
Got it. I was talking about the method name. But I get your point. Async vs/vs Sync.